vigor/forms.py

```diff
@@ -1,20 +1,57 @@
 from django import forms
-from .models import User
-from django.contrib.auth.forms import UserCreationForm
+from .models import UserPrefs, Meal
+from timezone_field import TimeZoneFormField
+
+class MealForm(forms.ModelForm):
+    submission_date = forms.DateField(input_formats = ['%d/%m/%Y', '%m/%d/%Y'], required = False)
+    meal_name = forms.CharField(required = False)
 
-class UserCreationFormExtended(UserCreationForm):
     def __init__(self, *args, **kwargs):
-        super(UserCreationFormExtended, self).__init__(*args, **kwargs)
-        self.fields['email'].required = True
-        self.fields['first_name'].required = True
-        self.fields['last_name'].required = True
+        date_format = kwargs.pop('date_format')
+        super(MealForm, self).__init__(*args, **kwargs)
+        self.fields['submission_date'].widget = forms.widget.DateInput(format = date_format)
+        self.fields['calories'].widget.attrs = {'placeholder': 320}
+
+    class Meta:
+        model = Meal
+        exclude = ()
 
-    def clean_email(self):
-        email = self.cleaned_data.get('email')
-        if email and User.objects.filter(email=email).count():
-            raise forms.ValidationError('This email is already bound to an existing profile')
-        return email
+class UserPrefsForm(forms.ModelForm):
+    time_zone = TimeZoneFormField()
 
     class Meta:
-        model = User
-        fields = ('username', 'email', 'first_name', 'last_name')
+        model = UserPrefs
+        fields = (
+            'gender',
+            'birth_year',
+            'birth_month',
+            'birth_day',
+            'ethnicity',
+            'height_feet',
+            'height_inches',
+            'weight',
+            'time_zone',
+            'date_format'
+        )
+
+
+class ProfileForm(forms.ModelForm):
+    class Meta:
+        model = UserPrefs
+        fields = ('gender', 'birth_year',)
+# from django.contrib.auth.forms import UserCreationForm
```

Comment:
@juliancoleman If there's no reason to keep this commented out stuff around, let's get rid of it! 😄